--- a/library/net_sockets.c	2018-12-21 07:20:49.000000000 -0500
+++ b/library/net_sockets.c	2019-01-12 15:30:20.026439300 -0500
@@ -34,7 +34,7 @@
 
 #if !defined(unix) && !defined(__unix__) && !defined(__unix) && \
     !defined(__APPLE__) && !defined(_WIN32) && !defined(__QNXNTO__) && \
-    !defined(__HAIKU__)
+    !defined(__HAIKU__) && !defined(__SWITCH__)
 #error "This module only works on Unix and Windows, see MBEDTLS_NET_C in config.h"
 #endif
 
@@ -90,6 +90,7 @@
 #include <fcntl.h>
 #include <netdb.h>
 #include <errno.h>
+#include <netinet/tcp.h>
 
 #define IS_EINTR( ret ) ( ( ret ) == EINTR )
 
@@ -176,6 +177,8 @@
 
         if( connect( ctx->fd, cur->ai_addr, MSVC_INT_CAST cur->ai_addrlen ) == 0 )
         {
+            int flag = 1;
+            setsockopt(ctx->fd, IPPROTO_TCP, TCP_NODELAY, (char *) &flag, sizeof(int));
             ret = 0;
             break;
         }
